core/src/main/java/org/mskcc/cbio/portal/servlet/CancerStudyView.java

```diff
@@ -221,20 +227,7 @@ private boolean buildResponse(HttpServletRequest request)
 	            studySampleMap.put(_study.getId(), updatedSampleList);
 	        	}
 	    });
-
-        // check if there are any studies in the response map, if no then check
-        if (studySampleMap.size() == 0) {
```

Comment:
We no longer differentiated causes?

Reply:
We are redirecting to home page if there are any `unKnownStudies`. There is no need to set any error attributes.

Reply:
Sorry! dont redirect to home page but show an error message which is at line
https://github.com/cBioPortal/cbioportal/pull/3631/files/0f02e898d60ac62ca986406a1d265f0795f9a976#diff-c17cf799714af30e11d56ccb957974f9R204

Reply:
OK, so we don't actually differentiate unknown or unauthorize studies anymore. It will make the logic much cleaner, but not sure whether this will confuse users.